packages/static-wado-webserver/lib/subscribers/dimse/operations/cfind.mjs

```diff
@@ -0,0 +1,80 @@
+import dcmjsDimse from "dcmjs-dimse";
+import dcmjs from "dcmjs";
+
+const { CFindRequest } = dcmjsDimse.requests;
+const { DicomMetaDictionary } = dcmjs.data;
+
+/**
+ * Returns a dcmjsdimse request for cfind.
+ * Based on request options it decides which request to return.
+ *
+ * @param {Object} requestOptions Object containing request's options.
+ * @returns
+ * @throws {Error} in case there is no valid combination of uids (on requestOptions) that represents a request
+ */
+export function createRequestFactory(requestOptions) {
```

Comment:
As part of the requestOptions, is there a way to provide information on what is already present?  That allows finer grained edits on the current retrieve status, and allows short circuiting the query.  There are 3 important shortcuts:
1. If the study doesn't exist at all, then C-Move the entire study.    Return "next" immediately to retrieve data on disk.
2. For each series, if the series doesn't exist, then C-move the series.  Do NOT perform instance level C-Find
3. For each series, if the number of series related instances matches, then assume up to date.
4. For each series, if the configuration is set to skip instance checks, then return, do not perform instance level C-Find

Reply:
WE gonna improve other cases later.